e2e/cypress/integration/channel/send_message_via_profile_popover_spec.js

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Profile popover', () => {
+    let newUser;
+    let currentChannelId;
+    let lastPostId;
+
+    before(() => {
+        // # Login as "user-1" and go to main channel view
+        cy.toMainChannelView('user-1');
+
+        // # Update user preferences
+        cy.apiSaveTeammateNameDisplayPreference('username');
+        cy.apiSaveMessageDisplayPreference();
+
+        // # Create new user and have it post a message
+        cy.getCurrentTeamId().then((teamId) => {
+            cy.getCurrentChannelId().then((id) => {
+                currentChannelId = id;
+            });
+
+            cy.createNewUser({}, [teamId]).then((user) => {
+                newUser = user;
+
+                const message = `Testing ${Date.now()}`;
+                cy.postMessageAs({sender: newUser, message, channelId: currentChannelId});
+
+                cy.waitUntil(() => cy.getLastPost().then((el) => {
+                    return el.find('.post-message__text > p')[0].textContent === message;
+                }));
+
+                cy.getLastPostId().then((postId) => {
+                    lastPostId = postId;
+                });
+            });
+        });
+    });
+
+    it('M19908 Send message in profile popover take to DM channel', () => {
+        // # On default viewport width of 1300
+        // # Click profile icon to open profile popover. Click "Send Message" and verify redirects to DM channel
+        verifyDMChanneViaSendMessage(lastPostId, '.profile-icon', newUser);
+
+        // # Click username to open profile popover. Click "Send Message" and verify redirects to DM channel
+        verifyDMChanneViaSendMessage(lastPostId, '.user-popover', newUser);
+
+        // # On mobile view
+        cy.viewport('iphone-6');
+
+        // # Click profile icon to open profile popover. Click "Send Message" and verify redirects to DM channel
+        verifyDMChanneViaSendMessage(lastPostId, '.profile-icon', newUser);
+
+        // # Click username to open profile popover. Click "Send Message" and verify redirects to DM channel
+        verifyDMChanneViaSendMessage(lastPostId, '.user-popover', newUser);
```

Comment:
For some reason, this test is failing after clicking on "Send Message" (on `cy.get('#channelIntro')` in line 88) because the DM for the user is not opening after the click. I haven't been able to figure out what's causing it, esp. because all the above tests pass.

Reply:
I thought it was fixed by the [referenced PR](https://github.com/mattermost/mattermost-webapp/pull/4142).  I made comment to fix it there - https://github.com/mattermost/mattermost-webapp/pull/4142/files#r344050408

Reply:
Updated the PR with the guard condition that should take care of the issue.

Reply:
Hi @saturninoabril 

I am seeing a different issue now when Cypress clicks on the user profile on RHS. It throws a JS error and does not opens up the profile popover. As a result, the test fails. Is this JS error anywhere related to PR referenced above?

![Screenshot 2019-11-08 at 10 27 09 PM](https://user-images.githubusercontent.com/1429138/68496105-95741d00-0277-11ea-8195-09f0cd30eb6e.png)

Reply:
What branch of webapp is that running? 

I noticed this morning there is another bug on daily related to not opening the profile popover but should not be related to my PR that has not been merged yet

Reply:
I have the latest master branch. So may be something is broken on the latest master and is related to the issue which you saw.

Reply:
I have raised this as a seperate issue 
https://mattermost.atlassian.net/browse/MM-19996

Looks like there was some PR merged recently causing some major breakages